src/main/kotlin/apply/application/ApplicantService.kt

```diff
@@ -3,21 +3,39 @@ package apply.application
 import apply.domain.applicant.Applicant
 import apply.domain.applicant.ApplicantRepository
 import apply.domain.applicant.Gender
+import apply.domain.applicant.dto.ApplicantRequest
+import apply.security.JwtTokenProvider
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
 import support.createLocalDate
 import javax.annotation.PostConstruct
 
 @Transactional
 @Service
-class ApplicantService(private val applicantRepository: ApplicantRepository) {
+class ApplicantService(
+    private val applicantRepository: ApplicantRepository,
+    private val jwtTokenProvider: JwtTokenProvider
+) {
     fun findAll(): List<Applicant> {
         return applicantRepository.findAll()
     }
 
     fun findByValue(value: String): List<Applicant> =
         applicantRepository.findByNameContainingOrEmailContaining(value, value)
 
+    fun validateOrCreateApplicantAndGenerateToken(applicantRequest: ApplicantRequest): String {
+        applicantRepository.findByEmail(applicantRequest.email)
+            ?.validate(applicantRequest)
+            ?: createApplicant(applicantRequest)
+
+        return jwtTokenProvider.createToken(applicantRequest.email)
```

Comment:
`applicantRequest.email` 대신 `applicant.email`을 사용하면 어떨까요?

```suggestion
        val applicant = applicantRepository.findByEmail(applicantRequest.email)
            ?.also { it.validate(applicantRequest) }
            ?: createApplicant(applicantRequest)

        return jwtTokenProvider.createToken(applicant.email)
```

Reply:
가려운데를 긁어주신 느낌이네요!
아직 범위지정함수에 익숙하지가 않은데, 감사합니다 👍